packages/fast-usdc/src/cli/lp-commands.js

```diff
@@ -0,0 +1,151 @@
+const parseNat = arg => {
+  try {
+    const n = Nat(BigInt(arg));
+    return n;
+  } catch {
+    throw new InvalidArgumentError('Not a number');
+  }
+};
+
+/** @param {string} arg */
+const parseDecimal = arg => {
+  try {
+    assertParsableNumber(arg);
+    const n = Number(arg);
+    return n;
+  } catch {
+    throw new InvalidArgumentError('Not a number');
+  }
+};
+
+/**
+ * @param {Command} program
+ * @param {{
+ *   fetch?: Window['fetch'];
+ *   vstorageKit?: Awaited<ReturnType<typeof makeVstorageKit>>;
+ *   stdout: typeof process.stdout;
+ *   stderr: typeof process.stderr;
+ *   env: typeof process.env;
+ *   now: typeof Date.now;
+ * }} io
+ */
+export const addLPCommands = (
+  program,
+  { fetch, vstorageKit, stderr, stdout, env, now },
+) => {
+  const loadVsk = async () => {
+    if (vstorageKit) {
+      return vstorageKit;
+    }
+    assert(fetch);
+    const networkConfig = await fetchEnvNetworkConfig({ env, fetch });
+    return makeVstorageKit({ fetch }, networkConfig);
+  };
+  /** @type {undefined | ReturnType<typeof loadVsk>} */
+  let vskP;
+
+  program
+    .command('deposit')
+    .description('Deposit USDC into pool')
+    .addHelpText(
+      'after',
+      '\nPipe the STDOUT to a file such as deposit.json, then use the Agoric CLI to broadcast it:\n  agoric wallet send --offer deposit.json --from gov1 --keyring-backend="test"',
+    )
+    .requiredOption('--amount <number>', 'USDC amount', parseDecimal)
+    .option('--offerId <string>', 'Offer id', String, `lpDeposit-${now()}`)
+    .action(async opts => {
+      vskP ||= loadVsk();
+      const vsk = await vskP;
+      /** @type {Brand<'nat'>} */
+      // @ts-expect-error it doesnt recognize usdc as a Brand type
+      const usdc = vsk.agoricNames.brand.USDC;
+      assert(usdc, 'USDC brand not in agoricNames');
+
+      const USDC_DECIMALS = 6;
+      const unit = AmountMath.make(usdc, 10n ** BigInt(USDC_DECIMALS));
+      const usdcAmount = multiplyBy(unit, parseRatio(opts.amount, usdc));
+
+      /** @type {OfferSpec} */
+      const offer = {
+        id: opts.offerId,
+        invitationSpec: {
+          source: 'agoricContract',
+          instancePath: ['fastUsdc'],
+          callPipe: [['makeDepositInvitation', []]],
+        },
+        proposal: {
+          give: {
+            USDC: usdcAmount,
+          },
+        },
+      };
+
+      /** @type {ExecuteOfferAction} */
+      const bridgeAction = {
+        method: 'executeOffer',
+        offer,
+      };
+
+      outputActionAndHint(bridgeAction, { stderr, stdout }, vsk.marshaller);
+    });
+
+  program
+    .command('withdraw')
+    .description("Withdraw USDC from the LP's pool share")
+    .addHelpText(
+      'after',
+      '\nPipe the STDOUT to a file such as withdraw.json, then use the Agoric CLI to broadcast it:\n  agoric wallet send --offer withdraw.json --from gov1 --keyring-backend="test"',
+    )
+    .requiredOption('--amount <number>', 'FastLP amount', parseNat)
```

Comment:
please change this amount likewise so that `--amount 1.00` denotes `AmountMath.make(usdc, 1_000_000n)`.

Reply:
Done